mat64/shadow.go

```diff
@@ -223,3 +225,56 @@ func rectanglesOverlap(off, aCols, bCols, stride int) bool {
 	// b strictly wraps and so must overlap with a.
 	return true
 }
+
+// trianglesOverlap returns whether the strided triangles a and b overlap
+// when b is offset by off elements after a but has at least one element before
+// the end of a. a and b have aSize and bSize respectively.
```

Comment:
Could you say that they're aSize x aSize, it's not clear in this context that aSize is a dimension and not a total number of elements.